row.py

```diff
@@ -209,49 +211,45 @@ def get_circles(item_path, output_path):
     )
 
 
-def ocr_images(directory):
-    """OCRs all image files (.jpg) in directory
+def get_characters(image):
+    """detect characters in an image
 
     Args:
+        image (np.array): The image to detect characters in
 
-         directory (Path): The directory where the images are located
-
-     Returns:
-         pd.dataframe: A dataframe with OCR results
+    Returns:
+        list: A list of detected characters
     """
-    #: create initial dataframe to work with
-    working_df = pd.DataFrame({"Number": [], "Filename": [], "Parcel": []})
-    working_df = working_df.astype({"Number": int, "Filename": object, "Parcel": object})
-
-    #: read all files in the directory
-    ocr_num = 0
-    file_list = list(directory.glob("*.jpg"))
-    for ocr_file in file_list:
-        ocr_count = ocr_file.stem.rsplit("_", 1)[1]
-
-        logging.debug("working on: %s", ocr_file.stem)
-        ocr_full_path = directory / ocr_file
-        #: read in image from bytes
-        byte_img = ocr_full_path.read_bytes()
-        ocr_img = cv2.imdecode(np.frombuffer(byte_img, dtype=np.uint8), 1)  # 1 means flags=cv2.IMREAD_COLOR
-
-        #: convert image to grayscale
-        ocr_gray = cv2.cvtColor(ocr_img, cv2.COLOR_BGR2GRAY)
-
-        ocr_text = pytesseract.image_to_string(ocr_gray, config=OCR_CONFIG)
-        ocr_text = ocr_text.replace("\n", "").replace("\t", "").replace(" ", "")
-        df_new = pd.DataFrame(
-            {
-                "Number": [int(ocr_count)],
-                "Filename": [ocr_file],
-                "Parcel": [ocr_text],
-            }
-        )
+    image_channels = cv2.imdecode(np.frombuffer(image, dtype=np.uint8), 1)  # 1 means flags=cv2.IMREAD_COLOR
```

Comment:
Are we calling np.frombuffer twice?  Once in row_cli.py before get_characters() is called (L92), and then again here inside the get_characters() function?  If we call it outside the function, I don't think we need it inside the function, too.

Reply:
good catch. I didn't like having numpy in the cli so let's keep it all in row and send in the bytes only.